### PR TITLE
fix(dynamodb-global): cannot deploy global tables due to unresolved resource dependencies

### LINKED ISSUE #4676: "Unresolved resource dependencies" when doing a CDK Deployment of Global DynamoDB Table
As detailed in this bug report: https://github.com/aws/aws-cdk/issues/2900  , there was an issue where it was not possible to create a DynamoDB Global Table in CDK with a stack as the scope - it must be the top-level application that is used for the scope. This was supposedly fixed in #2900 / CDK 1.14.0, and the issue that was previously there IS gone, but now I have a new error when I attempt to deploy:

----
 ❌  GlobalDataTesttableCustomResourceD8874B5D failed: ValidationError: Template format error: Unresolved resource dependencies [tableGlobalTableuswest1B808DC21, tableGlobalTableuseast20418E1CB] in the Resources block of the template
Template format error: Unresolved resource dependencies [tableGlobalTableuswest1B808DC21, tableGlobalTableuseast20418E1CB] in the Resources block of the template
----

No issues are encountered when running cdk synth.

Please note I am using CDK 1.14.0, and python 3.7. I've attached the files which cause this problem to occur. If the declaration of "ddbg.GlobalTable" in data_stack.py uses "scope" instead of "self", this problem goes away (but this is counter to the way scope is usually used in every CDK example I've seen).

### Reproduction Steps

**app.py**
```python
#!/usr/bin/env python3
from aws_cdk import core
from service import data_stack

app = core.App()
east = {'region':'us-east-2'}
west = {'region':'us-west-1'}
data_stack = data_stack.DataStack(app, 'GlobalDataTest', regions=[east['region'], west['region']], env=east)
app.synth()
```

###############

**data_stack.py**
```python
from aws_cdk import core
from aws_cdk import aws_dynamodb as ddb
from aws_cdk import aws_dynamodb_global as ddbg

class DataStack(core.Stack):
    def __init__(self, scope: core.Construct, id: str, regions: list, **kwargs) -> None:
        super().__init__(scope, id, **kwargs)
        self.table_name = 'globaldatatest.global.table'
        self.table = ddbg.GlobalTable(
            self, # FAILURE OCCURS DUE TO THIS LINE - putting "scope" works
            'table',
            partition_key={'name': 'key', 'type': ddb.AttributeType.NUMBER},
            table_name=self.table_name,
            regions=regions,
        )
```

###############

### Error Log

 ❌  GlobalDataTesttableCustomResourceD8874B5D failed: ValidationError: Template format error: Unresolved resource dependencies [tableGlobalTableuswest1B808DC21, tableGlobalTableuseast20418E1CB] in the Resources block of the template
Template format error: Unresolved resource dependencies [tableGlobalTableuswest1B808DC21, tableGlobalTableuseast20418E1CB] in the Resources block of the template

### Environment

  - **CLI Version      :1.16.219**
  - **Framework Version:1.14.0**
  - **OS               :OS X 10.15**
  - **Language         :python 3.6**

### Other

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by skinny85**
The DynamoDB global table construct was incorrectly declaring its dependencies,
having the coordinator stack depend on the resources in the table stacks,
instead of on the stacks themselves.
That resulted in resources in the coordinator stack depending on the global tables,
which cannot work, as those are from different stacks.
Changed the logic to declare dependencies between the stacks explicitly.

Fixes #4676

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-dynamodb-global/test/integ.dynamodb.global.expected.json [hunk 1]
```diff
@@ -1,4 +1,6 @@
 [
+  {
+  },
```
Comment (eladb) [inline packages/@aws-cdk/aws-dynamodb-global/test/integ.dynamodb.global.expected.json]: What’s going on here?

**Comment by skinny85** [comment]
> Also, the PR description says that the dependencies were not correctly defined but it doesn’t explain what was the incorrect definition.

Updated.

> PR title for bugs should describe the bug and not the solution. The title is used in Changelog under “fixes” so it should speak in terms of the bug that was fixed and not the solution.

That's what I tried to do; the bug was an incorrect dependency declaration, but it manifested itself with being unable to use global DynamoDB tables with `Stack` as the parent.

If you have a suggestion for a better title, I'm all ears.

File: packages/@aws-cdk/aws-dynamodb-global/test/integ.dynamodb.global.expected.json (hunk 1 above)
Comment (skinny85) [reply packages/@aws-cdk/aws-dynamodb-global/test/integ.dynamodb.global.expected.json]: This is empty, because I added a `Stack` to the test which doesn't contain any resources. If you think it will be clearer, I can add something like a `WaitConditionHandle` to it and add it to the expectations.

Comment (eladb) [reply packages/@aws-cdk/aws-dynamodb-global/test/integ.dynamodb.global.expected.json]: oh, interesting...

Comment (eladb) [reply packages/@aws-cdk/aws-dynamodb-global/test/integ.dynamodb.global.expected.json]: That's fine.

**Comment by eladb** [comment]
> If you have a suggestion for a better title, I'm all ears.

Usually the title of the bug (https://github.com/aws/aws-cdk/issues/4676) is a good title for the fix. Maybe rephrased and shortened a little:

```
fix(dynamodb-global): cannot deploy global tables due to unresolved resource dependencies
```

**Comment by skinny85** [comment]
> > If you have a suggestion for a better title, I'm all ears.
> 
> Usually the title of the bug (#4676) is a good title for the fix. Maybe rephrased and shortened a little:
> 
> ```
> fix(dynamodb-global): cannot deploy global tables due to unresolved resource dependencies
> ```

Changed.
